### DISCUSSION TITLE
Nested Value in default values variable.

### DISCUSSION THREAD

**Discussion by writes-of-spring** [react-hook-form/react-hook-form#6119] [Q&A]
When trying to store useForm default values as a variables as they are received as props, and can change, using NestedValue<T[]> returns 

Type '{ regionId: string; name: string; }[]' is not assignable to type 'NestedValue<Region[]>'.
  Property '[$NestedValue]' is missing in type '{ regionId: string; name: string; }[]' but required in type '{ [$NestedValue]: never; }'.

example of code: 
```typescript 
type Region = {
  regionId: string;
  name: string;
};
type FormValues = {
  regions: NestedValue<Region[]>;
};

const defaultValues: FormValues = {
  regions: [{ regionId: "Strawberry", name: "strawberry" }]
};
const regions: Region[] = [
  { regionId: "Chocolate", name: "chocolate" },
  { regionId: "Strawberry", name: "strawberry" },
  { regionId: "Vanilla", name: "vanilla" }
];

export default function App() {
  const {
    register,
    handleSubmit,
    setValue,
    formState: { errors }
  } = useForm<FormValues>({
    defaultValues
  });
```

CSB Link: 
https://codesandbox.io/s/react-hook-form-nestedvalue-forked-d1yie?file=/src/App.tsx:309-860

Any pointers in the right direction would be greatly appreciated

**Comment by tparvi** [comment]
I ran into this same issue. Did you manage to figure it out?

**Comment by Elfayer** [comment]
Why not just type it as follow?

```ts
type FormValues = {
  regions: Region[];
};
```

Thing is, I don't think you can register a List, but only primitives as `regions.regionsId`. Your `Autocomplete` component should also be wrapped by a `<Controller>`, see: https://react-hook-form.com/advanced-usage#ControlledmixedwithUncontrolledComponents

If you need a list in your form, then you need a FieldArray: https://react-hook-form.com/api/usefieldarray

**Comment by tparvi** [comment]
I think our case is a bit of an **edge case**. So we have react-select which user can use to choose items into that array but we only want to validate the array as a whole. So we are not using `useFieldArray `and we only want to get single error message (`FieldError `instead of `FieldErrors`) during validation. This is why we tried to use `NestedValue `but it seems you cannot actually create new `NestedValue<T>` type. Unfortunately we are mapping from api model to form model somewhere else and we would need that.

So my solution to get around is to nest that array inside another value like this:

```
type PersonContainer = {
  persons: Person[]
}

type FormValues = {
  persons: PersonContainer
}

```
This should allow use to validate only persons (one error message for that field).

**Reply by Elfayer** [reply]
Again, I don't think you should be selecting an "object" but rather something like an `id`. From there nothing prevents you from having a simple variable in your react component as:
```ts
const [regionId] = useWatch<FormValues>({ control, name: 'regionId' });
const regionSelected = regions.find(region => region.id === regionId)
```
